Update server.md (#832)
pi-hole/docs: docs/guides/vpn/wireguard/server.md

> ## Set your Pi-hole to listen on all interfaces

Reviewer: ```suggestion
## Set your Pi-hole to allow only local requests
```
